test-jest/lib/authorization-schemes.unit.test.js

```diff
@@ -152,97 +152,98 @@ describe('authorization-schemes', () => {
+        await expect(authorizationSchemes.licenceScheme().payload(request, getMockH())).resolves.toEqual('response')
 
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalled()
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB123CD')
+        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB12 3CD')
         expect(request.app.authorization.contactId).toEqual('12345')
       })
 
       it('should return an error and auth should not be present if LicenceApi.getContactFromLicenceKey call fails', async () => {
-        const request = {
-          payload: {
-            licence: '123456',
-            postcode: 'AB123CD'
-          },
-          app: {}
-        }
-        const h = {
-          continue: 'response'
-        }
+        const request = getMockLicenceRequest()
         LicenceApi.getContactFromLicenceKey.mockImplementation(() => {
           throw new Error()
         })
 
-        await expect(authorizationSchemes.licenceScheme().payload(request, h)).rejects
+        await expect(authorizationSchemes.licenceScheme().payload(request, getMockH())).rejects
 
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalled()
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB123CD')
+        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB12 3CD')
         expect(request.app.authorization).toBeUndefined()
       })
 
       it('should return h.continue and auth should not be present if LicenceApi.getContactFromLicenceKey call fails with a statusCode beginning with 4', async () => {
-        const request = {
-          payload: {
-            licence: '123456',
-            postcode: 'AB123CD'
-          },
-          app: {}
-        }
-        const h = {
-          continue: 'response'
-        }
+        const request = getMockLicenceRequest()
         LicenceApi.getContactFromLicenceKey.mockImplementation(() => {
           throw new ResponseError.Error('Error', ResponseError.status.UNAUTHORIZED)
         })
 
-        await expect(authorizationSchemes.licenceScheme().payload(request, h)).resolves.toEqual('response')
+        await expect(authorizationSchemes.licenceScheme().payload(request, getMockH())).resolves.toEqual('response')
 
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalled()
-        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB123CD')
+        expect(LicenceApi.getContactFromLicenceKey).toHaveBeenCalledWith(request, '123456', 'AB12 3CD')
         expect(request.app.authorization).toBeUndefined()
       })
+
+      it.each([
+        ['ba21nw', 'BA2 1NW'],
+        [' AB12    3CD ', 'AB12 3CD'],
+        ['AB123CD ', 'AB12 3CD']
+      ])('formats the UK postcode %s successfully as %s', async (postcode, replacedValue) => {
```

Comment:
Could you also add a scenario for `A9 9AA` https://en.wikipedia.org/wiki/Postcodes_in_the_United_Kingdom#Validation.

This is a really good idea, but do all postcodes in CRM have a space before the last 3 characters? In GAFL I saw you say that the business-lib automatically adds the space, but what about when you add the address manually? Also in the business-lib there is an overseas postcode validator, does it do the same if you enter an overseas postcode? It may be worth asking Ravi to do some sort of query about the postcode data in CRM.

Reply:
Any postcodes that come through GAFL (which is the vast majority of them) will have the space added, whether the address comes through the postcode lookup or is manually entered.

The only chance we have for a postcode without a space is if a record is edited in the CRM interface, although missing a space here will also break other features like the easy renewal lookup.